ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2620,18 +2622,18 @@ public static void listFilesInsideAcidDirectory(Path acidDir, FileSystem srcFs,
     }
   }
 
-  private void listFilesCreatedByQuery(Path loadPath, long writeId, int stmtId, boolean isInsertOverwrite,
-      List<Path> newFiles) throws HiveException {
+  private List<FileStatus> listFilesCreatedByQuery(Path loadPath, long writeId, int stmtId) throws HiveException {
```

Comment:
Can we return List<String> to avoid extra map(FileStatus::getPath) in the method usages?

Reply:
No, we need the filestatuses, so we can reuse them later for populating the quickstats. In a follow up patch I will try to do the other way around and collect FileStatus everywhere in loadPartition instead of Path, because writeNotification could also benefit having the filestatuses ready. But it is not that trivial change in the non direct insert cases.

Reply:
gotcha!